src/Utils.php

```diff
@@ -9,6 +9,8 @@ function must_have_idx<Tk, Tv>(?KeyedContainer<Tk, Tv> $arr, Tk $idx): Tv {
   $result = idx($arr, $idx);
   invariant(
     $result !== null,
+    /* HH_IGNORE_ERROR[4110] - HHVM 3.18+ enforces \HH\FormatString, ignoring pending upstream documentation. */
+    /* HH_IGNORE_ERROR[4027] - HHVM 3.18+ enforces \HH\FormatString, ignoring pending upstream documentation. */
```

Comment:
what you want is:

```
invariant(
   $result !== null,
   'Index %s not found in container',
   $idx,
);
```

Given that Tk isn't constrained, the code as-is could error at runtime (including on < 3.18) if `$idx` contains `%s`